File: app/src/main/java/io/memetic/alcohdex/core/dagger/modules/AppModule.java
package io.memetic.alcohdex.core.dagger.modules;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.res.Resources;

import com.google.gson.Gson;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.memetic.alcohdex.App;
import io.memetic.alcohdex.R;

import static android.content.Context.MODE_PRIVATE;

/**
 * TODO class description
 *
 * @author manuel.silva@mobileforming.com
 *         4/17/17
 */
@Module
public class AppModule {

    private final App mApp;
    private final Resources mResources;
    private SharedPreferences mSharedPreferences;

    public AppModule(App app) {
        mApp = app;
        mResources = app.getResources();
        mSharedPreferences = mApp.getSharedPreferences(
                mApp.getString(R.string.preference_file_key), MODE_PRIVATE
        );
    }

    @Provides
    @Singleton
    public Context provideContext() {
        return mApp;
    }


    @Provides
    @Singleton
    public SharedPreferences provideSharedPrefs() {
        return mSharedPreferences;
    }

    @Provides
    @Singleton
    public Resources provideResources() {
        return mResources;
    }

    @Provides
    @Singleton
    public Gson provideGson() {
        return new Gson();
    }
}

File: app/src/main/java/io/memetic/alcohdex/core/dagger/modules/AppModule.kt
package io.memetic.alcohdex.core.dagger.modules

import android.content.Context
import android.content.Context.MODE_PRIVATE
import android.content.SharedPreferences
import android.content.res.Resources
import com.google.gson.Gson
import dagger.Module
import dagger.Provides
import io.memetic.alcohdex.App
import io.memetic.alcohdex.R
import javax.inject.Singleton

/**
 * TODO class description

 * @author manuel.silva@mobileforming.com
 * *         4/17/17
 */
@Module
class AppModule(private val mApp: App) {
    private val mResources: Resources = mApp.resources
    private val mSharedPreferences: SharedPreferences = mApp.getSharedPreferences(
            mApp.getString(R.string.preference_file_key), MODE_PRIVATE
    )

    @Provides
    @Singleton
    fun provideContext(): Context {
        return mApp
    }


    @Provides
    @Singleton
    fun provideSharedPrefs(): SharedPreferences {
        return mSharedPreferences
    }

    @Provides
    @Singleton
    fun provideResources(): Resources {
        return mResources
    }

    @Provides
    @Singleton
    fun provideGson(): Gson {
        return Gson()
    }
}
